### DISCUSSION TITLE
Table: how to remove a record that the policy specify the user should not view

### DISCUSSION THREAD

**Discussion by bvicini** [filamentphp/filament#14638] [Help] [answered]
### Package

Table builder

### Package Version

v3.2.121

### How can we help you?

I have a policy that specifies when a user can "handle" a given model. When listing in the resource table, all models appears: how can i prefilter the table on the base of the policy? Ty

**Comment by ronssij** [comment, marked as answer]
@bvicini It should not be on the policy. It should be on a model or on the query level.

**Reply by bvicini** [reply]
In the model? But we write the policies precisely so that we don't have to write the authorization logic in the model. I solved the problem by modifying the query, but in several cases it would be useful if a user who does not have permission to see a model does not see it in the table at all. for example when, due to privacy laws, the user should not even know about the existence of that record. Ty for yor answer

**Reply by gregoryloichot** [reply]
You can use GlobalScope to hide record for the user.

**Reply by bvicini** [reply]
Ty, i will try that :) See you
